WordPress/gutenberg
# feat: Update Image block failed upload visuals

## Related PRs

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/6491
* https://github.com/wordpress-mobile/WordPress-iOS/pull/22282
* https://github.com/wordpress-mobile/WordPress-Android/pull/19878

## What?
Display a custom message for failed media uploads that are paused due to the lack of an internet connection. 

## Why?
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6407. Improve the UX by better communicating to users why the media upload is failed so that they may take action. 

## How?
Integrate a new "paused" media upload state that is sent by the host app. For block using `enablePausedUploads` (currently, only Image blocks), a specific "Waiting for connection" error message is displayed. 

## Testing Instructions




1. Add an Image, Gallery, or Media + Text block. 
2. Disable your devices network connection (e.g. airplane mode). 
3. Attempt to attach image(s) to the block. 
4. Verify the "Waiting for connection" error message is displayed. 
5. Repeat steps 1–4 changing when the network connection is disabled, e.g., midway through an upload. 

### Testing Instructions for Keyboard

Attempt to complete the aforementioned testing instructions via a mobile device's screen reader. 

## Screenshots or screencast 

<details><summary>iOS</summary>

| Light mode | Dark mode |
| - | - |
| ![iOS image block failed upload design in light mode](https://github.com/WordPress/gutenberg/assets/438664/7c5c407e-9b16-4e97-8333-54bbfa774602) | ![iOS image block failed upload design in dark mode](https://github.com/WordPress/gutenberg/assets/438664/0c049af8-9316-4871-bd8c-ec90ff49181b) |

</details>

<details><summary>Android</summary>

| Light mode | Dark mode |
| - | - |
| ![Android image block failed upload design in light mode](https://github.com/WordPress/gutenberg/assets/438664/3e2e8b9f-a542-49df-bfba-31a1ea2afacd) | ![Android image block failed upload design in dark mode](https://github.com/WordPress/gutenberg/assets/438664/fb33247d-68fd-485c-9a91-7cb8c319f05c) |

</details>
